connection.ts

```diff
@@ -609,9 +609,5 @@ export class Connection {
     await this.bufWriter.write(terminationMessage);
     await this.bufWriter.flush();
     this.conn.close();
-    delete this.conn;
-    delete this.bufReader;
-    delete this.bufWriter;
-    delete this.packetWriter;
```

Comment:
Are these no longer needed?

Reply:
Now I see #165, fantastic, thanks!

Reply:
The new TS version stipulates that non-null asserted values cannot be `delete`-ed.

I assumed that if the `close` method was called, the code could assume that those properties will no longer be accessed. As they were asserted non-null, it's not like the compiler was checking for the case that they've been deleted anyway, so if app-space code is messing with them directly, that's on the developer.

If a larger refactor is palatable, we could make those class members optional rather than asserted non-null, and add in the proper checks. Then, the `delete` calls could be re-added.

Reply:
Thanks. Thanks a good point - 🤷 - not sure such a change adds much value. 😬